File: 205-isomorphic-strings/isomorphic-strings.kt
class Solution {
    fun isIsomorphic(s: String, t: String): Boolean {
        if(s.length != t.length) return false

        var map : MutableMap<Char,Char> = mutableMapOf()

        for(i in 0 until s.length){
            if(map[s[i]] == null){
                if(map.containsValue(t[i])) return false
                map[s[i]] = t[i]
            }
            else{
                if(map[s[i]] != t[i]) return false
            }
        }

        return true
    }
}

File: 205-isomorphic-strings/isomorphic-strings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        m = dict()
        for sc,tc in zip(s,t):
            if sc in m:
                if m[sc] != tc:
                    return False
            elif tc in m.values():
                return False
            m[sc] = tc
        return True 